### ISSUE TITLE
Paring fails on Boolean types with a value of 0 or 1

### ISSUE THREAD

**Issue by stefanc82** [SeanConnelly/Cogs#5]
When a JSON string contains a boolean field with a value of 0 or 1, the parser does not recognise it as valid boolean values.

**Comment by SeanConnelly** [comment]
Values 0 and 1 are not JSON boolean values.
If strictly going by the specification then these should be false and true.
However, if the target property is a boolean, then I think there is room to be less strict on the type conversion. I will open up the logic to accept 0 and 1, but only if the target is a boolean.
I also noticed that my unit tests don't inject any whitespace into the JSON strings and found that your previous example was including the whitespace in front of these 0 and 1 values. This is now fixed.
I will make and push this change today.
Sean.
